### PR TITLE
Fix various failing tests

### PR DESCRIPTION
**Pull request by aminya**

### CODE REVIEW

File: test/unit/socket-draft-dgram-test.ts [hunk 1]
```diff
@@ -5,60 +5,61 @@ import {assert} from "chai"
...
+    }
 
-      beforeEach(function () {
-        dgram = new draft.Datagram()
-      })
+    let dgram: draft.Datagram
 
-      afterEach(function () {
-        dgram.close()
-        global.gc?.()
```
Comment (rotu) [inline test/unit/socket-draft-dgram-test.ts:19]: Good! It makes me uncomfortable that we're calling `global.gc` all over the place without any seeming justification.

**Comment by rotu** [comment]
I tried to give targeted and actionable feedback but it seems to have been ignored. @aminya, @bluca, in the spirit of open source, how can I better contribute to this project?

**Comment by aminya** [comment]
@rotu I think you have misread the code. I didn't remove the `.gc` calls. They were just indented. Changing the GC calls is out of the scope of this PR.

**Comment by rotu** [comment]
> @rotu I think you have misread the code. I didn't remove the .gc calls. They were just indented. Changing the GC calls is out of the scope of this PR.

@aminya oops - you're right you didn't remove them. BTW, these calls make the garbage collector run *synchronously*, so if any of the tests are at all timing-sensitive, you might notice some weirdness, especially since things are running in parallel now.

**Comment by aminya** [comment]
@rotu Some of the remaining failing tests are related to GC, but we have to fix them. Take a look at Issue 466 for context.

**Pull request merged**